packages/dds/map/src/test/mocha/summary.spec.ts

```diff
@@ -0,0 +1,197 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import { existsSync, readFileSync, writeFileSync } from "node:fs";
+import path from "node:path";
+import {
+	MockContainerRuntimeFactory,
+	MockFluidDataStoreRuntime,
+	MockStorage,
+} from "@fluidframework/test-runtime-utils";
+import { convertSummaryTreeToITree } from "@fluidframework/runtime-utils";
+import { DirectoryFactory, SharedDirectory } from "../..";
+import { assertEquivalentDirectories } from "./directoryEquivalenceUtils";
+
+// Simple filter to avoid tests with a name that would accidentally be parsed as directory traversal or other confusing things.
+const nameCheck = new RegExp(/^[^"/\\]+$/);
+
+assert(__dirname.match(/dist[/\\]test[/\\]mocha$/));
+const snapshotsFolder = path.join(__dirname, `../../../src/test/mocha/snapshots`);
+assert(existsSync(snapshotsFolder));
+
+let currentTestName: string | undefined;
+let currentTestFile: string | undefined;
+
+function useSnapshotDirectory(dirPath: string = "/"): void {
+	const normalizedDir = path.join(snapshotsFolder, dirPath);
+	// Basic sanity check to avoid bugs like accidentally recursively deleting everything under `/` if something went wrong (like dirPath navigated up directories a lot).
+	assert(normalizedDir.startsWith(snapshotsFolder));
+
+	beforeEach(function (): void {
+		currentTestName = this.currentTest?.title ?? assert.fail();
+		currentTestFile = path.join(normalizedDir, `${currentTestName}.json`);
+	});
+
+	afterEach(() => {
+		currentTestFile = undefined;
+		currentTestName = undefined;
+	});
+}
+
+interface TestScenario {
+	only?: boolean;
+	skip?: boolean;
+	name: string;
+	runScenario: () => SharedDirectory;
+	// Utilized to test the back-compat of snapshots, i.e. ensuring the ability to load and collaborate
+	// in the old format documents even though we no longer write in this format
```

Comment:
```suggestion
	/**
	 * Whether running the scenario produces a snapshot which matches the saved one.
	 * This is used to test back-compat of snapshots, i.e. ensuring current code can load older documents.
	 * @remarks - It may be valuable to confirm clients can collaborate on such documents
	 * after loading them.
	 */
```